spirv_glsl.cpp

```diff
@@ -9606,6 +9606,9 @@ string CompilerGLSL::access_chain_internal(uint32_t base, const uint32_t *indice
 	bool pending_array_enclose = false;
 	bool dimension_flatten = false;
 
+	// If we are translating access to a structured buffer, the first subscript '._m0' must be hidden
+	const bool hide_first_subscript = count > 1 && is_user_type_structured(base) ? 1 : 0;
```

Comment:
Seems more elegant to just `indices++; count--;` here and jump ahead in the type hierarchy instead of adding deeper branching later in the code.

Reply:
Can you please provide a suggestion how jumping to the type hierarchy should look like since there are several variables that track certain states such as `row_major_matrix_needs_conversion`?
While the first subscript must be hidden in the output, the type of `._m0` must still be picked up during the loop:
```cpp
type = &get<SPIRType>(type->member_types[index]);
```

Reply:
Ok, I didn't consider that. I guess this can be kept then.